### PR TITLE
feat(inputtext): add new inputText component

### PR DESCRIPTION
**Pull request by mdjmoore**
## Summary

**Structural Element (Gene)**
Shortcut ticket: [sh-157938](https://app.shortcut.com/sci-design-system/story/157938/implement-inputtext)
- add text input component to code repo and story book
- component name: InputText
- sdsType = Text Field / Text Area
- sdsStage = Default / User Input
- states: default, hover, active, disabled

## Checklist

- [x] Default Story in Storybook
- [x] LivePreview Story in Storybook
- [x] Test Story in Storybook
- [x] Tests written
- [x] Variables from `defaultTheme.ts` used wherever possible
- [x] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: src/core/InputText/index.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    <FormControl>
+      <InputText
+        id="test"
+        sdsType="textField"
+        label="Label"
+        hideLabel={false}
+        placeholder="Value"
+        data-testid="inputTextBase"
+      />
+      {/* @ts-ignore */}
```
Comment (mdjmoore) [inline src/core/InputText/index.stories.tsx:130]: Typescript (rightfully) flag this as an error, but I want to test that failure state, so ignore TS errors on the following line

**Comment by tihuan** [comment]
@clarsen-czi I resolved your Chromatic comments for Meagan, since she doesn't have owner access to resolve those 😆 PTAL Chromatic again thank you!!

File: src/core/InputText/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      `Error: czifui component InputText requires id and label props for accessibility.`
+    );
+    return <></>;
+  }
+
+  return (
+    <>
+      {!hideLabel && <StyledLabel htmlFor={id}>{label}</StyledLabel>}
+      <StyledInputBase
+        {...rest}
```
Comment (tihuan) [inline src/core/InputText/index.tsx:55]: should we move the `..rest` to the bottom to allow users to override the props 😄 ? For example, `type` could be `password`, `color`, `date`, etc..

https://www.w3schools.com/html/html_form_input_types.asp

Thank you!

**Review by tihuan — APPROVED** [review]
Looks super good! Thanks for figuring out `InputText` component, @mdjmoore 🙏 🌟 🚀 !!

**Pull request merged**